### DISCUSSION TITLE
Does the schema language support polymorphism?

### DISCUSSION THREAD

**Discussion by marlenekoh** [authzed/spicedb#1496] [Q&A] [answered]
I'm trying to model jira software and jira service management projects in a schema. I have read https://authzed.com/blog/user-defined-roles

My simplified schema is something like this:
```
definition user {}

definition jira_project {
  relation viewer: user
  relation editor: user

  permission create_issue: editor
  permission view_issue: viewer + editor
}

definition jsm_project {
  relation viewer: user
  relation editor: user

  permission create_issue: editor
  permission view_issue: viewer + editor
  permission approve_issue: editor
}
```

I'm wondering if it's possible to do something like this?
```
definition user {}

definition project {
  relation viewer: user
  relation editor: user

  permission create_issue: editor
  permission view_issue: viewer + editor
}

definition jira_project extends project {}

definition jsm_project extends project {
  permission approve_issue: editor
}
```
or if there's any easier way to do this? Thanks in advance 😄

**Comment by vroldanbet** [comment, marked as answer]
@marlenekoh 👋🏻 there is no support for composition or inheritance at the moment, unfortunately, so I'm afraid for now you'll have to go the verbose route of copy/pasting. Feel free to open a feature request!
